tasks/main.yml

```diff
@@ -48,6 +48,35 @@
   ansible.builtin.set_fact:
     pve_kvm_inferred_vmid: "{{ pve_kvm_clone_newid if(pve_kvm_clone_newid is defined and pve_kvm_clone_from_existing) else ( pve_kvm_vmid if(pve_kvm_vmid is defined) else pve_kvm_extracted_vmid.stdout ) }}"
 
+# Loop over pve_kvm_net_interfaces and ensure they have been created and are up to date
+- name: Ensure KVM NICs have been created or are up to date
+  community.general.proxmox_nic:
+    api_host: "{{ pve_api_host }}"
+    api_user: "{{ pve_api_user | default(omit) }}"
+    api_password: "{{ pve_api_password | default(omit) }}"
+    api_token_id: "{{ pve_api_token_id | default(omit) }}"
```

Comment:
Token authentication is not working in this role. This bug [has been solved for the collection](https://git.interior.edu.uy/cielito/proxmox/-/merge_requests/1), so this freature still more deserves to be worked there.

Reply:
I'll try to get some time to implement this PR in our collection! Whilst, do not hesitate to test it, it has several improvements from here.